### PR TITLE
Rename `__pyccel__` and `__epyccel__` folders

### LINKED ISSUE #969: Deactivate FileLock for tests
**Describe the bug**
FileLock is used to prevent compilation problems when running pyccel in parallel. However the creation of locking files could potentially be slow. We wish to add a mode to pyccel which allows it to be run without locks. This mode could then be used when we know pyccel is being run in serial. It can additionally be used for running tests by creating folders named `__pyccel__${PYTEST_XDIST_WORKER}` where `${PYTEST_XDIST_WORKER}` is an environment variable containing the name of the thread defined by pytest. This may increase the speed of the tests by removing 0.05s waits between testing for the availability of a lock

### PR DESCRIPTION
**Pull request by EmilyBourne**
Change name of `__pyccel__` and `__epyccel__` folders to avoid collisions during tests. The new name is the old name with an additional suffix defined as the contents of the environment variable `PYTEST_XDIST_WORKER` (when the variable is not present there is no suffix). The tests now run around 10mins faster. Fixes #969 

Add a test for the `--out` argument

**Bot fixes**
- Remove review handler from old bot
- Correct error leading to tests being triggered twice when PR is marked as ready
- Correct error where labels are not removed when PR is made draft
- Correct documentation comparison folder
- Correct spelling relative directory
- Improve documentation

### CODE REVIEW

File: docs/compiler.md [hunk 1]
```diff
@@ -66,5 +66,5 @@ By using the expected compiler, the expected libraries are correctly linked.
 
 Pyccel offers the `--conda-warnings` flag which takes one of the following options : `off`, `basic`, `verbose`.
 This controls the visibility of Conda-related warnings. The default value is `basic` which indicates when folders are ignored.
-The `verbose` option additionally outputs a list of all ignored folders. These folders can be un-ignored using a user-defined compiler file as described above.
+The `verbose` option additionally outputs a list of all ignored folders. If these folders souldn't be ignored then this can be avoided by using a user-defined compiler file as described above.
```
Comment (yguclu) [inline docs/compiler.md:69]: Typo: souldn't -> shouldn't (maybe "should not" is better)

File: docs/quickstart.md [hunk 2]
```diff
@@ -451,6 +451,11 @@ We are also working on supporting [MPI](https://en.wikipedia.org/wiki/Open_MPI),
 
 In the future we plan to support GPU programming with [CUDA](https://en.wikipedia.org/wiki/CUDA) and [task-based parallelism](https://en.wikipedia.org/wiki/Task_parallelism).
 
+## Cleaning up the environment
+
+Pyccel generates various files, in order to help clean up the environment after using it, we therefore also provide the command line tool: `pyccel_clean`.
```
Comment (yguclu) [inline docs/quickstart.md:456]: Typo: pyccel_clean -> pyccel-clean

File: docs/compiler.md (hunk 1 above)
Comment (EmilyBourne) [reply docs/compiler.md]: Fixed 2ecc2f18

File: docs/quickstart.md (hunk 2 above)
Comment (EmilyBourne) [reply docs/quickstart.md]: Fixed 131cb2b7

File: pyccel/commands/pyccel_clean.py [hunk 3]
```diff
@@ -13,33 +13,39 @@
...
+    path_dir : str, default = current working directory
+        The path to the folder which should be cleaned.
+
+    recursive : bool, default = True
+        Indicates whether the function should recurse into
+        sub-folders.
+
+    remove_shared_libs : bool, default = False
+        Indicates whether shared libraries generated by
+        python should also be removed from the directory
```
Comment (yguclu) [inline pyccel/commands/pyccel_clean.py:35]: Please capitalize Python 😜

File: pyccel/commands/pyccel_clean.py [hunk 4]
```diff
@@ -13,33 +13,39 @@
 ext_suffix = sysconfig.get_config_var('EXT_SUFFIX')
 
 def pyccel_clean(path_dir = None, recursive = True, remove_shared_libs = False, remove_programs = False):
-    """ Remove __pyccel__ and __epyccel__ folders as well
-    as any python shared libraries from the directory path_dir
+    """
+    Remove folders generated by Pyccel.
+
+    Remove `__pyccel__X` and `__epyccel__X` folders as well
+    as any python shared libraries from the directory path_dir.
```
Comment (yguclu) [inline pyccel/commands/pyccel_clean.py:20]: Here too!

File: pyccel/commands/pyccel_clean.py (hunk 3 above)
Comment (EmilyBourne) [reply pyccel/commands/pyccel_clean.py]: Fixed 5915651a

File: pyccel/commands/pyccel_clean.py (hunk 4 above)
Comment (EmilyBourne) [reply pyccel/commands/pyccel_clean.py]: Fixed https://github.com/pyccel/pyccel/commit/5915651aad51cda962992e37cf59da37ed89ecc2

File: ci_tools/summarise_pyspelling.py [hunk 5]
```diff
@@ -78,15 +78,17 @@ def find_all_words(file_path, search_word):
...
+        with open(os.path.join(pyccel_folder,'.dict_custom.txt'), encoding="utf-8") as d:
             internal_dict = [w.strip() for w in d.readlines()]
 
         with open(args.output, 'w', encoding="utf-8") as f:
             print("There are misspelled words", file=f)
             for name, words in errors.items():
                 print("## `", name, "`", file=f)
                 for w in words:
-                    words_list = find_all_words(os.path.join("..",name.strip(":")), w)
+                    words_list = find_all_words(os.path.join(pyccel_folder,name.strip(":")), w)
```
Comment (yguclu) [inline ci_tools/summarise_pyspelling.py:91]: Would you mind adding an empty space after the commas? This is needed once in each of the 3 lines you modified, inside the call to `join`.

File: pyccel/codegen/pipeline.py [hunk 6]
```diff
@@ -159,7 +159,7 @@ def handle_error(stage):
         folder = os.path.abspath(folder)
 
     # Define directory name and path for pyccel & cpython build
-    pyccel_dirname = '__pyccel__'
+    pyccel_dirname = '__pyccel__'+os.environ.get('PYTEST_XDIST_WORKER', '')
```
Comment (yguclu) [inline pyccel/codegen/pipeline.py:162]: Could you please add empty spaces around the `+` binary operator?

File: pyccel/codegen/pipeline.py [hunk 7]
```diff
@@ -320,7 +320,7 @@ def handle_error(stage):
     # ...
     # Determine all .o files and all folders needed by executable
     def get_module_dependencies(parser, deps):
-        mod_folder = os.path.join(os.path.dirname(parser.filename), "__pyccel__")
+        mod_folder = os.path.join(os.path.dirname(parser.filename), '__pyccel__'+os.environ.get('PYTEST_XDIST_WORKER', ''))
```
Comment (yguclu) [inline pyccel/codegen/pipeline.py:323]: Same here

File: pyccel/epyccel.py [hunk 8]
```diff
@@ -83,13 +83,11 @@ def get_unique_name(prefix, path):
...
 
-    save_path = os.path.join(path, '__epyccel__')
-
     # Create new directories if not existing
-    os.makedirs(save_path, exist_ok=True)
+    os.makedirs(path, exist_ok=True)
 
     # Ensure that the name is not in use by another thread
-    lock = FileLock(os.path.join(save_path, module_name)+'.lock')
+    lock = FileLock(os.path.join(path, module_name)+'.lock')
```
Comment (yguclu) [inline pyccel/epyccel.py:90]: Could you add some empty spaces around the `+` binary operator?

File: pyccel/epyccel.py [hunk 9]
```diff
@@ -169,22 +167,38 @@ def epyccel_seq(function_or_module, *,
...
+        dirpath = os.path.dirname(function_or_module.__file__)
+
+    # Define working directory 'folder'
+    if folder is None:
+        folder = os.path.dirname(dirpath)
+    else:
+        folder = os.path.abspath(folder)
+
+    # Define directory name and path for epyccel files
+    epyccel_dirname = '__epyccel__'+os.environ.get('PYTEST_XDIST_WORKER','')
```
Comment (yguclu) [inline pyccel/epyccel.py:186]: Could you add some empty spaces around the `+` binary operator? And one after the comma?

File: tests/pyccel/test_pyccel.py [hunk 10]
```diff
@@ -46,7 +46,7 @@ def get_exe(filename, language=None):
 def insert_pyccel_folder(abs_path):
     base_dir = os.path.dirname(abs_path)
     base_name = os.path.basename(abs_path)
-    return os.path.join(base_dir, "__pyccel__", base_name)
+    return os.path.join(base_dir, "__pyccel__"+os.environ.get('PYTEST_XDIST_WORKER', ''), base_name)
```
Comment (yguclu) [inline tests/pyccel/test_pyccel.py:49]: Could you add some empty spaces around the `+` binary operator?

**Review by yguclu — CHANGES_REQUESTED** [review]

File: tests/pyccel/test_pyccel.py (hunk 10 above)
Comment (EmilyBourne) [reply tests/pyccel/test_pyccel.py]: Fixed b245ece3

File: pyccel/epyccel.py (hunk 9 above)
Comment (EmilyBourne) [reply pyccel/epyccel.py]: Fixed b245ece3

File: pyccel/codegen/pipeline.py (hunk 6 above)
Comment (EmilyBourne) [reply pyccel/codegen/pipeline.py]: Fixed b245ece3

File: pyccel/epyccel.py (hunk 8 above)
Comment (EmilyBourne) [reply pyccel/epyccel.py]: Fixed b245ece3

File: ci_tools/summarise_pyspelling.py (hunk 5 above)
Comment (EmilyBourne) [reply ci_tools/summarise_pyspelling.py]: Fixed [93140d4](https://github.com/pyccel/pyccel/pull/1451/commits/93140d47fbf43b86d413f50f8b116e56e1daead3)

File: tests/pyccel/test_pyccel.py (hunk 10 above)
Comment (EmilyBourne) [reply tests/pyccel/test_pyccel.py]: If you care about these spaces, should we have a pep8 linter?

Comment (yguclu) [reply tests/pyccel/test_pyccel.py]: I do not know. The PEP8 might be too aggressive when we align things vertically for readability.

**Review by yguclu — APPROVED** [review]

**Comment by yguclu** [comment]
Can you please list all PR changes in the PR description?

**Comment by yguclu** [comment]
Can you please explain in the PR description what is the new name of the generated folders in the different cases?

**Comment by EmilyBourne** [comment]
> Can you please list all PR changes in the PR description?
> Can you please explain in the PR description what is the new name of the generated folders in the different cases?

Done

**Pull request merged**